src/components/molecules/VenueChat/VenueChat.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useContext, useEffect, useState, useMemo, FC } from "react";
+import { ChatContext } from "components/context/ChatContext";
+import "./CampChat.scss";
+import { useUser } from "hooks/useUser";
+import { useSelector } from "hooks/useSelector";
+import { useVenueId } from "hooks/useVenueId";
+import { getDaysAgoInSeconds } from "utils/time";
+import { VENUE_CHAT_AGE_DAYS } from "settings";
+import ChatBox from "components/molecules/Chatbox";
+import useRoles from "hooks/useRoles";
+
+interface ChatOutDataType {
+  messageToTheBand: string;
+}
+
+const VenueChat: FC = () => {
+  const venueId = useVenueId();
+  const { userRoles } = useRoles();
+  const { user } = useUser();
+  const { chats, venue } = useSelector((state) => ({
+    chats: state.firestore.ordered.venueChats,
+    venue: state.firestore.data.currentVenue,
+  }));
```

Comment:
As above RE: extracting selectors, particularly about avoiding this 'compound selector' style

Reply:
this is still an issue

Ideal fix:

Create the selectors outside of the component (eg. top of file, `src/utils/selectors.ts`, etc):

```typescript
const currentVenueSelector = (state: RootState) => state.firestore.data.currentVenue
const chatsSelector = (state: RootState) => state.firestore.ordered.venueChats
```

Then you can do:
```suggestion
  const venue = useSelector(currentVenueSelector)
  const chats = useSelector(chatsSelector)
```

---

I believe`currentVenue` is one of the things I have refactored in https://github.com/sparkletown/sparkle/pull/721 / https://github.com/sparkletown/sparkle/pull/726 
Though don't let that block this PR. Fix as above, and I can worry about any fixes my branch will need later.